netlify/next-runtime
# fix: add missing data to middleware request object

### Summary

When the middleware runs on the edge, it's missing data that is hydrated by the NextServer when run in the context of the origin server.

Also fixes which property is accessed on nextURL in the context of the MiddlewareRequest.

### Test plan

[Test site](https://app.netlify.com/sites/ep-next-enhanced-middleware-test)

Can also do the following:
* pull this branch
* Update the `next.config.js` of the middleware demo to include the following:
  
```
  i18n: {
    locales: ["en-US"],
    defaultLocale: "en-US",
  }
```

* Add a console.log below [this line](https://github.com/netlify/next-runtime/blob/5cf8c52ead9a9013906201697a6a4a32fd8368f7/demos/middleware/middleware.ts#L8) referencing the `locale` and `defaultLocale`
* Confirm that the values are `en-US`

**Note:** It's a known issue that the middleware example isn't working entirely as expected and are not related to these changes. For instance, the path to set the `netlifyCookie` via the 'Cookies API' route isn't setting the cookie properly, although the user will be routed to the correct page. Issue for this has been created [here](https://github.com/netlify/next-runtime/issues/1639)

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal
Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/190

### Standard checks:




---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.
